src/nspawn/nspawn.c

```diff
@@ -4667,6 +4674,33 @@ static int run_container(
         /* Normally redundant, but better safe than sorry */
         (void) kill(*pid, SIGKILL);
 
+        if (arg_private_network) {
+                /* Move network interfaces back to the parent network namespace. */
+                _cleanup_close_ int parent_netns_fd = -1;
+
+                r = namespace_open(getpid_cached(), NULL, NULL, &parent_netns_fd, NULL, NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to open parent network namespace: %m");
+
+                r = safe_fork(NULL, FORK_RESET_SIGNALS|FORK_DEATHSIG|FORK_WAIT|FORK_REOPEN_LOG, NULL);
```

Comment:
please add FORK_LOG, so that the call logs about errors on its own. (in your current patch on failure we don't log at all here!)

Also, please drop FORK_REOPEN_LOG. There's not really a need to do that (reopening the logs makes sense if we migrate cgroups or so, i.e. our client side process props change and we want the logging service to take notice, or when we do some fd tricks and want the log fd to go away, but neither is the case here)